constraints.pro

```diff
@@ -131,100 +149,119 @@ gen_enforced_field(WorkspaceCwd, 'description', DescriptionWithoutTrailingPeriod
+  workspace_field(WorkspaceCwd, 'repository.url', RepoUrl),
+  repo_name(RepoUrl, RepoName),
+  atomic_list_concat(['https://github.com/MetaMask/', RepoName, '/issues'], CorrectBugsUrl).
+% Non-published packages do not have a bugs URL.
+gen_enforced_field(WorkspaceCwd, 'bugs.url', null) :-
   workspace_field(WorkspaceCwd, 'private', true).
 
-% "license" must be "MIT" for all workspace packages and unset for the root.
+% All packages must specify Git as the repository type.
+gen_enforced_field(WorkspaceCwd, 'repository.type', 'git').
+
+% All packages must match the URL of a repo within the MetaMask organization.
+gen_enforced_field(WorkspaceCwd, 'repository.url', 'https://github.com/MetaMask/<insert repo name here>.git') :-
+  workspace_field(WorkspaceCwd, 'repository.url', RepoUrl),
+  \+ repo_name(RepoUrl, _).
+% The repository URL for non-root packages must match the same URL used for the
+% root package.
+gen_enforced_field(WorkspaceCwd, 'repository.url', RepoUrl) :-
+  workspace_field('.', 'repository.url', RepoUrl),
+  WorkspaceCwd \= '.'.
+
+% The license for all published packages must be MIT.
 gen_enforced_field(WorkspaceCwd, 'license', 'MIT') :-
   \+ workspace_field(WorkspaceCwd, 'private', true).
+% Non-published packages do not have a license.
 gen_enforced_field(WorkspaceCwd, 'license', null) :-
   workspace_field(WorkspaceCwd, 'private', true).
 
-% "main" must be "dist/index.js" for workspace packages and unset for the
-% root.
+% The entrypoint for all published packages must be the same.
 gen_enforced_field(WorkspaceCwd, 'main', './dist/index.js') :-
   \+ workspace_field(WorkspaceCwd, 'private', true).
+% Non-published packages must not specify an entrypoint.
 gen_enforced_field(WorkspaceCwd, 'main', null) :-
   workspace_field(WorkspaceCwd, 'private', true).
 
-% "types" must be "dist/index.d.ts" for workspace packages and unset for the
-% root.
+% The type definitions entrypoint for all publishable packages must be the same.
 gen_enforced_field(WorkspaceCwd, 'types', './dist/index.d.ts') :-
   \+ workspace_field(WorkspaceCwd, 'private', true).
+% Non-published packages must not specify a type definitions entrypoint.
 gen_enforced_field(WorkspaceCwd, 'types', null) :-
   workspace_field(WorkspaceCwd, 'private', true).
 
-% "files" must be ["dist/"] for workspace packages and [] for the root.
+% The list of files included in published packages must be everything in `dist/`.
 gen_enforced_field(WorkspaceCwd, 'files', ['dist/']) :-
   \+ workspace_field(WorkspaceCwd, 'private', true).
+% The root package must specify an empty set of published files. (This is
+% required in order to be able to import anything in development-only scripts,
+% as otherwise the `node/no-unpublished-require` ESLint rule will disallow it.)
 gen_enforced_field(WorkspaceCwd, 'files', []) :-
-  workspace_field(WorkspaceCwd, 'private', true).
+  WorkspaceCwd = '.'.
+% All other non-published packages must not specify a set of published files.
```

Comment:
Hmm. Not sure about this; assigning `files` is useful even for non-published packages, for distinguishing between "test" and "non-test" files. This distinction is used by some of our lint rules.

Reply:
Ah okay! Removed in[dab1717](https://github.com/MetaMask/controllers/pull/1032/commits/dab1717430904bb30f1c99e61d1a8e2e6d32c031).